packages/playwright-core/src/server/registry/index.ts

```diff
@@ -701,12 +707,9 @@ export class Registry {
       throw new Error(`ERROR: Playwright does not support ${descriptor.name} on ${hostPlatform}`);
     if (hostPlatform === 'generic-linux' || hostPlatform === 'generic-linux-arm64')
       logPolitely('BEWARE: your OS is not officially supported by Playwright; downloading Ubuntu build as a fallback.');
-    const downloadHost =
-        (downloadHostEnv && getFromENV(downloadHostEnv)) ||
-        getFromENV('PLAYWRIGHT_DOWNLOAD_HOST') ||
-        'https://playwright.azureedge.net';
     const downloadPath = util.format(downloadPathTemplate, descriptor.revision);
-    const downloadURL = `${downloadHost}/${downloadPath}`;
+    const customHostOverride = (downloadHostEnv && getFromENV(downloadHostEnv)) || getFromENV('PLAYWRIGHT_DOWNLOAD_HOST');
+    const downloadURLs = PLAYWRIGHT_CDN_MIRRORS.map(mirror => `${customHostOverride || mirror}/${downloadPath}`) ;
```

Comment:
This would imply 3 attempts for the same custom Host url. Do we want this?

I'd just have an array of a single url if there's a custom host.